src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,73 @@
+package duke.task;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructor for Task.
+     *
+     * @param description description for the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns mark status in String format.
+     *
+     * @return mark status
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    /**
+     * Marks the mark status.
+     *
+     */
+    public void setMark() {
+        isDone = true;
+    }
+
+    /**
+     * Unmarks the mark status.
+     *
+     */
+    public void setUnMark() {
+        isDone = false;
+    }
+
+    /**
+     * Returns true if the key given is contained in
+     * the description of the task.
+     *
+     * @param key the string key need to be contained.
+     * @return a boolean whether the key is contained.
+     */
```

Comment:
consider renaming this method to is... as it returns a boolean.

Reply:
Okay, thanks for the suggestion!